### DISCUSSION TITLE
how to query by multiple vertex ids?

### DISCUSSION THREAD

**Discussion by jbockle** [Gremlinq/ExRam.Gremlinq#624] [Q&A] [answered]
sorry I think this should be simple but I can't find/figure out a solution - if I have a list of ids for a vertex, how do I pass that to my query

e.g. something like
```csharp
var ids = new [] { "foo", "bar };
var persons = await g.V<Person>(ids).ToArrayAsync();
```

I get an error from cosmos regarding the keys
```
ExceptionMessage :
        Gremlin query syntax error: Invalid composite key argument for g.V(...). Composite key requires two values [partition_key, id], per composite key expression. For example: g.V([pk0, id0], [pk1, id1], ...).
        Context : graphcompute
        Scope : graphstg-logplan
        GraphInterOpStatusCode : QuerySyntaxError
        HResult : 0x80131500
```

**Comment by danielcweber** [comment, marked as answer]
Look a the signature of `V`: It takes `params object[]`. What you're passing is a array of strings, which will be treated as a single object. So you are effectively passing an array of strings as key, which won't work. `var ids = new object[] { "foo", "bar };` will do the trick.

**Reply by jbockle** [reply]
dang I knew it was easy.. thanks
